alcs-frontend/src/app/features/application/overview/overview.component.ts

```diff
@@ -1,15 +1,106 @@
+import { ApplicationMeetingService } from '../../../services/application/application-meeting/application-meeting.service';
+import { ApplicationDetailedDto, ApplicationDto } from '../../../services/application/application.dto';
+import { TimelineEvent } from '../../../shared/timeline/timeline.component';
 
 @Component({
   selector: 'app-overview',
   templateUrl: './overview.component.html',
-  styleUrls: ['./overview.component.scss']
+  styleUrls: ['./overview.component.scss'],
 })
 export class OverviewComponent implements OnInit {
+  private application?: ApplicationDetailedDto;
+  public events: TimelineEvent[] = [];
 
-  constructor() { }
+  constructor(
+    private applicationDetailService: ApplicationDetailService,
+    private meetingService: ApplicationMeetingService
+  ) {}
 
   ngOnInit(): void {
+    this.applicationDetailService.$application
+      .pipe(
+        tap((app) => {
+          if (app) {
+            this.meetingService.fetch(app.fileNumber);
+          }
+        })
+      )
+      .pipe(combineLatestWith(this.meetingService.$meetings))
+      .subscribe(([application, meetings]) => {
+        if (application) {
+          this.application = application;
+          this.events = this.mapApplicationToEvents(application, meetings);
+        }
+      });
   }
 
+  mapApplicationToEvents(application: ApplicationDto, meetings: ApplicationMeetingDto[]): TimelineEvent[] {
+    //TODO: Add other meeting types once their pages are ready
+    const editLink = new Map<string, string>([['IR', './info-request']]);
+
+    const mappedEvents: TimelineEvent[] = [];
+    if (application.dateReceived) {
+      mappedEvents.push({
+        name: 'Submitted to ALC',
+        startDate: new Date(application.dateReceived),
+        isFulfilled: true,
+      });
+    }
+
+    if (application.dateAcknowledgedIncomplete) {
+      mappedEvents.push({
+        name: 'Acknowledged Incomplete',
+        startDate: new Date(application.dateAcknowledgedIncomplete),
+        isFulfilled: true,
+      });
+    }
+
+    if (application.dateAcknowledgedComplete) {
+      mappedEvents.push({
+        name: 'Acknowledged Complete',
+        startDate: new Date(application.dateAcknowledgedComplete),
+        isFulfilled: true,
+      });
+    }
+
+    if (application.decisionMeetings.length) {
+      const events: TimelineEvent[] = application.decisionMeetings.map((meeting, index) => ({
+        name: application.decisionMeetings.length === 1 ? `Decision Meeting` : `Decision Meeting #${index + 1}`,
+        startDate: new Date(meeting.date),
+        isFulfilled: true,
+      }));
+      mappedEvents.push(...events);
+    }
+
+    if (application.decisionDate) {
+      mappedEvents.push({
+        name: 'Decision Made',
+        startDate: new Date(application.decisionDate),
+        isFulfilled: true,
+      });
+    }
+
+    meetings.sort((a, b) => a.startDate - b.startDate);
```

Comment:
I can see that bellow you are using getTime for sorting is intentional for omitting it here?

Reply:
These are still numbers from the API, I convert them to Dates while dolng the mapping.